ocs_ci/ocs/ui/validation_ui.py

```diff
@@ -565,3 +566,18 @@ def check_capacity_breakdown(self, project_name, pod_name):
             logger.error(f"The pod {pod_name} not found on capacity_breakdown")
             res = False
         return res
+
+    def validate_unprivileged_access(self):
+        """
+        Function to verify the unprivileged users can't access ODF dashbaord
+        """
+        self.do_click(self.validation_loc["developer_dropdown"])
+        self.do_click(self.validation_loc["select_administrator"], timeout=5)
+        try:
+            self.navigate_odf_overview_page()
+        except TimeoutException as ex:
+            logger.info(
+                f"As expected, ODF dashboard is not available for the unprivileged user: {ex}"
+            )
+        else:
+            assert False, "Unexpected, unprivileged users can access ODF dashboard"
```

Comment:
if there is no exception in try block, else block will execute, which means navigation under try block worked, but then else block gets executed where you are asserting false, so the test will always fail in this scenario.

Reply:
Yes correct that is what expected. Only criteria for test success is try block throws exception.

Reply:
what I can do is basically, instead of assert here... like you suggested raise a custom exception which is then caught in the test function and that can be used to assert the test.